eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -212,10 +212,28 @@ jobs:
       # Download and unzip target specific tests
       - template: /eng/pipelines/common/download-artifact-step.yml
         parameters:
-          unpackFolder: '$(managedTestArtifactRootFolderPath)'
           artifactFileName: '$(managedTestArtifactName)$(archiveExtension)'
           artifactName: '$(managedTestArtifactName)'
           displayName: 'managed test artifacts (built on ${{ parameters.managedTestBuildOsGroup }}${{ parameters.managedTestBuildOsSubgroup }})'
+          ${{ if ne(parameters.testBuildPhased, true) }}:
+            # These are the only tests unpack directly
+            unpackFolder: '$(managedTestArtifactRootFolderPath)'
+          ${{ if eq(parameters.testBuildPhased, true) }}:
+            # We want the specific tests to overlay the generic tests
+            # Unpacking directly casues issues due to not handling overwriting existing files
+            # Unpack to a temp folder and copy below
+            unpackFolder: '$(Build.SourcesDirectory)/__unpack__/$(managedTestArtifactName)'
+
+      - ${{ if eq(parameters.testBuildPhased, true) }}:
+        # Copy target specific tests on top of the generic test
+        # Expected to overwrite duplicates
```

Comment:
Would it make sense to instead port into this PR building the target specific tests as part of the run job, or is that a very big change?

Reply:
That PR already exists #35645.  If it is right it should fix the issue.

It is a bigger change, so it may take longer to review/merge.

Reply:
Ok, since this is a temp workaround, if it fixes the issue I don't want to block on this, so fine with me and we can remove this workaround on #35645